x/leverage/types/weighted_coin.go

```diff
@@ -95,27 +93,46 @@ func (wsp WeightedSpecialPair) before(b WeightedSpecialPair) bool {
 }
 
 // Add returns the sum of a weightedDecCoins and an additional weightedDecCoin.
-// The result is sorted.
+// The result is sorted, has any duplicate denoms combined, and any zero amounts omitted.
```

Comment:
This wasn't causing any bugs since positions are created from borrower collateral and borrows, which are unique in state.

The improvement does make the struct more resilient to unintended behavior if it ever occurs though.